cmd/compute-domain-kubelet-plugin/device_state.go

```diff
@@ -181,6 +184,20 @@ func (s *DeviceState) Unprepare(ctx context.Context, claimRef kubeletplugin.Name
 	// Rely on local checkpoint state for ability to clean up.
 	checkpoint := newCheckpoint()
 	if err := s.checkpointManager.GetCheckpoint(DriverPluginCheckpointFileBasename, checkpoint); err != nil {
+		// Some errors returned by `GetCheckpoint` are permanent. Some of them
+		// may be retryable. For now, handle an explicit set of errors as
+		// permanent, and treat every other error as retryable.
+
+		if errors.Is(err, &json.UnmarshalTypeError{}) {
```

Comment:
Note: there is no overarching error type representing any error during `Unmarshal()`.

As of https://pkg.go.dev/encoding/json#UnmarshalTypeError, `Unmarshal()` can actually return three different types of error:

> Unmarshal returns an InvalidUnmarshalError.

> Unmarshal returns a SyntaxError.

> Unmarshal returns an UnmarshalTypeError

`SyntaxError` means "bad JSON", we also want to treat that as permanent.

What would be a concise way to test "is this error type any of these...?"